packages/online-editor/src/companionFs/CompanionFsHooks.ts

```
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

About these hooks, would they perform duplicated actions if multiple tabs of the app were open in the browser?

---

Yes :)

---

For it not to happen, we would need a Shared worker. I opted to make them ignore errors, in case another tab was faster.

---

Maybe a TODO here or some inline comments? 😁

---

I'll create an issue for this, ok?